langchain_tools_demo/main.py

```diff
@@ -19,15 +19,16 @@
 
 import uvicorn
 from fastapi import Body, FastAPI, HTTPException, Request
-from fastapi.responses import HTMLResponse, PlainTextResponse
+from fastapi.responses import PlainTextResponse, RedirectResponse
 from fastapi.staticfiles import StaticFiles
 from fastapi.templating import Jinja2Templates
-from langchain.agents.agent import AgentExecutor
 from markdown import markdown
 from starlette.middleware.sessions import SessionMiddleware
 
 from agent import init_agent, user_agents
 
+GOOGLE_REDIRECT_URI = "http://localhost:8081"
```

Comment:
I think this needs up be updated to be dynamic -- this changes when we are deployed to cloud run

Reply:
set to use referer url